src/context/meta-tags-context.php

```diff
@@ -528,20 +528,19 @@ public function generate_schema_page_type() {
 				$type = 'CollectionPage';
 				break;
 			default:
-				$additional_type = $this->indexable->schema_page_type;
-				if ( \is_null( $additional_type ) ) {
-					$additional_type = $this->options->get( 'schema-page-type-' . $this->indexable->object_sub_type );
+				$additional_type = $this->get_additional_type();
+				$type            = [ 'WebPage' ];
+				if ( ! \is_null( $additional_type ) ) {
+					$type = [ 'WebPage', $additional_type ];
 				}
 
-				$type = [ 'WebPage', $additional_type ];
-
 				// Is this indexable set as a page for posts, e.g. in the WordPress reading settings as a static homepage?
 				if ( (int) \get_option( 'page_for_posts' ) === $this->indexable->object_id ) {
 					$type[] = 'CollectionPage';
 				}
 
-				// Ensure we get only unique values, and remove any null values and the index.
-				$type = \array_filter( \array_values( \array_unique( $type ) ) );
+				// Ensure we get only unique values.
+				$type = \array_values( \array_unique( $type ) );
```

Comment:
Can you explain why we do not filter out `null` values anymore here?

Reply:
Because `null` will not occur here as we now check whether `$additional_type` is null before adding it to `$type`.